np.sh

```diff
@@ -1,8 +1,18 @@
 #!/usr/bin/env sh
+
+if test 0 -ne `git status --porcelain | wc -l`; then
+	echo "Unclean working tree. Commit or stash changes first." >&2;
+	exit 128;
+fi
+
+if test "00" -ne `git fetch; git rev-list --count --left-right "@{upstream}"...HEAD | awk '{ print $1$2 }'`; then
```

Comment:
Yeah, either one works, I just chose the verbose option because it was clearer, but I can go with the terse one if you prefer.